x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/evaluator/predicate/operator/comparison/InMapper.java

```diff
@@ -40,10 +41,10 @@ public ExpressionEvaluator.Factory map(In in, Layout layout) {
             ExpressionEvaluator.Factory eqEvaluator = ((ExpressionMapper) EQUALS).map(eq, layout);
             listEvaluators.add(eqEvaluator);
         });
-        return dvrCtx -> new InExpressionEvaluator(dvrCtx.blockFactory(), listEvaluators.stream().map(fac -> fac.get(dvrCtx)).toList());
+        return dvrCtx -> new InExpressionEvaluator(dvrCtx, listEvaluators.stream().map(fac -> fac.get(dvrCtx)).toList());
```

Comment:
`drvCtx.blockFactory()` where the driverContext is ThrowingDriverContext can bring down Elasticsearch nodes. This can happen during the rolling upgrade from 8.11 to 8.12+ (see #103830).